File: challenge8/challenge8.py
# Create a function `calculateInconsistentSum(arr)` that takes an integer array as 
# input and returns a single integer between 0-50.

def challenge8(arr):
    odd_sum = 0;
    even_mult = 1;
    
    for i in range(len(arr)):
        if i % 2 == 0:
            even_mult *= arr[i]
        else:
            odd_sum += arr[i]
    
    return abs(odd_sum - even_mult) % 51

def main():
    # Get input from stdin
    arr = list(map(int, input().split()))
    # Call function and print result
    print(challenge8(arr))

main()

File: challenge8/challenge8.c
// Create a function `calculateInconsistentSum(arr)` that takes an integer array as 
// input and returns a single integer between 0-50.

#include <stdlib.h>

int challenge8(int* arr, int size) {
    int[] oddSum = 0;
    int evenMult = 1;
    
    for (int i = 0; i < size; i++) {
        if (i % 2 == 0) {
            evenMult *= arr[i];
        } else {
            oddSum += arr[i];
        }
    }
    
    return abs(oddSum - evenMult) % 51;
}

int main() {
    // Get input from stdin
    int n;
    scanf("%d", &n);
    int arr[n];
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    // Call the challenge function
    int result = challenge8(arr, n);
    printf("%d\n", result);
    return 0;
}